### PR TITLE
subsys: tracing: add RAM backend

### PR DESCRIPTION
**Pull request by julien-massot**
While trying to investigate on threads scheduling I noticed that I'm lacking I/O to output
tracing data, then I decide to output the data to a RAM buffer that I retrieved later using gdb.

Please pickup this patch if you think it could makes sense for other platforms too !

### CODE REVIEW

**Comment by nashif** [comment]
@julien-massot  thank you for the patch, can you add some documentation of how this can be used with gdb in doc/guides/tracing/. Would be appreciated.

**Comment by julien-massot** [comment]
> @julien-massot thank you for the patch, can you add some documentation of how this can be used with gdb in doc/guides/tracing/. Would be appreciated.

Sure will do

File: doc/guides/tracing/index.rst [hunk 1]
```diff
@@ -182,6 +183,20 @@ the tracing data::
 The resulting CTF output can be visualized using babeltrace or TraceCompass
 by pointing the tool to the ``data`` directory with the metadata and trace files.
 
+Using RAM backend
+=================
+
+For devices that do not available I/O for tracing such as USB or UART but have
```
Comment (nashif) [inline doc/guides/tracing/index.rst:189]: that do not have ...

Comment (julien-massot) [reply doc/guides/tracing/index.rst]: thanks !

**Review by nashif — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Pull request merged**